tests/console/snapper_cleanup_timeline.pm

```diff
@@ -0,0 +1,51 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Verify that secure boot is set as expected.
+# Maintainer: Sofia Syrianidou <ssyrianidou@suse.com>
+
+use strict;
+use warnings;
+use base 'y2_installbase';
+use testapi;
+use scheduler 'get_test_suite_data';
+use Test::Assert ':all';
+
+sub pre_run_hook {
+    my $test_data = get_test_suite_data();
+    select_console 'root-console';
+    record_info("Configuration", "Configure snapper.");
+    foreach my $param (keys %{$test_data->{snapper_config}}) {
+        assert_script_run("snapper set-config $param=$test_data->{snapper_config}->{$param}", fail_message => "Snapper configuration failed for parameter $param");
+    }
+}
+
+sub run {
+    record_info("Check quota", "Verify that the percentage of root filesystem quota is less than 50%");
+    my $disk_space   = script_output("df -h /  | awk \'NR==2 {print \$4}\'");
+    my $qgroup_space = script_output("btrfs qgroup show -F --sync / | awk \'NR==3 {print \$2}\'");
+    die "Root filesystem quota take more than 50% of root disk space" if ($qgroup_space >= 0.5 * $disk_space);
```

Comment:
From the logs:
```
Argument "34G" isn't numeric in multiplication (*) at /var/lib/openqa/share/tests/sle/tests/console/snapper_cleanup_timeline.pm line 33.
Argument "2.22GiB" isn't numeric in numeric ge (>=) at /var/lib/openqa/share/tests/sle/tests/console/snapper_cleanup_timeline.pm line 33.
```

Reply:
Yes, these are warnings. Is that a problem?

Reply:
it could be any side effect, comparing strings as numeric...IMO, also they are not using the same unit, but that could be overlooked.

Reply:
I changed the unit of btrfs show to GB. Thanks for noticing. About comparing strings as numeric, from what I understand so far, it's typical perl behavior to proceed with the first part of string that looks like number, but if you have had experience with similar issues I can check if I can use some library like Str2num. I tried sprint "%0.2f" , values become "5.32" instead of "5.32GB" but remain as strings and warning messages are still printed.

Reply:
Perl does the type conversion and remove the ending part so the unit are not a problem in this case and the operations are done correctly  both  (1) comparison of strings with numeric operator and (2) multiplication of numeric by string  but the warning is still there unless deactivated. Remove warnings helps to dissect problems in a test with less polluted logs so we should try whenever is possible.
I was trying with this script where not warnings appears when the unit are removed despite being still strings:
```perl
#! /usr/bin/perl -w

use strict;
use warnings;

my $disk_space = "34";
my $qgroup_space = "2.22";

my $result = ($qgroup_space >= 0.5 * $disk_space) ? "true" : "false";
print "$result\n";

```

Reply:
my suggestion to try to resemble that automatic perl conversion avoiding the warning afterward would be this:
```perl
my $convert_numeric = qr/[^0-9.]+/;

$disk_space =~ s/$convert_numeric//;
$qgroup_space =~ s/$convert_numeric//;
```

Reply:
Weird, this does not produce warning but if I use sprintf, it  does. Done.